guide.md

```diff
@@ -0,0 +1,406 @@
+Getting Started 2.0
+===
+
+For this guide, we're going to walk through some simple examples for each of the
+pieces of CRUD. Each step in this guide will build on the previous, and is meant
+to be followed along. This guide assumes that you're using PostgreSQL on
+Rust nightly. We'll talk about how to use Diesel on stable Rust in the final
+chapter. Before we start, make sure you have PostgreSQL installed and running.
+
+The first thing we need to do is generate our project.
+
+```shell
+cargo new diesel_demo
+cd diesel_demo
+```
+
+First, let's add Diesel to our dependencies. We're also going to use a tool
+called [`.env`][dotenv-rust] to manage our environment variables for us. We'll
+add it to our dependencies as well.
+
+[dotenv-rust]: https://github.com/slapresta/rust-dotenv
+
+```toml
+[dependencies]
+diesel = "0.5.0"
+diesel_codegen = { version = "0.5.0", default-features = false, features = ["nightly", "postgres"] }
+dotenv = "0.8.0"
+dotenv_macros = "0.8.0"
+```
+
+Diesel provides a separate [CLI][diesel-cli] tool to help manage your project.
+Since it's a standalone binary, and doesn't affect your project's code directly,
+we don't add it to `Cargo.toml`. Instead, we just install it on our system.
+
+[diesel-cli]: https://github.com/sgrif/diesel/tree/master/diesel_cli
+
+```shell
+cargo install diesel_cli
+```
+
+We need to tell Diesel where to find our database. We do this by setting the
+`DATABASE_URL` environment variable. On our development machines, we'll likely
+have multiple projects going, and we don't want to pollute our environment. We
+can put the url in a `.env` file instead.
+
+```shell
+echo DATABASE_URL=postgres://localhost/diesel_demo > .env
+```
+
+Now Diesel CLI can set everything up for us.
+
+```shell
+diesel setup
+```
+
+This will create our database for us (if it didn't already exist), and create an
+empty migrations directory that we can use to manage our schema (more on that
+later).
+
+Now we're going to write a small CLI that lets us manage a blog (ignoring the
+fact that we can only access the database from this CLI...). The first thing
+we're going to need is a table to store our posts. Let's create a migration for
+that:
+
+```shell
+diesel migration generate create_posts_table
+```
+
+You should see output which looks something like this:
+
+```
+Creating migrations/20160202154039_create_posts/up.sql
+Creating migrations/20160202154039_create_posts/down.sql
+```
+
+Migrations allow us to evolve the database schema over time. Each migration can
+be applied (`up.sql`) or reverted (`down.sql`). Applying and immediately
```

Comment:
Applied and reverted are not antonyms. I wonder if there are two words here we could use that are true antonyms of each other.